### PR TITLE
Shiftability work

### PR DESCRIPTION
**Pull request by ethteck**
Changes to the linker script + yaml to support shiftability + some matches

### CODE REVIEW

**Comment by BowserSlug** [comment]
🚀🚀🚀🚀🚀🚀 This PR matches 6 functions (+0.09%) on `us`.

⚠️ This PR introduces 2 warnings:
- src/main.c:64: warning: implicit declaration of function `func_8002CA00'
- src/os/39cb0_len_100.c:10: warning: redundant redeclaration of `nuIdleFunc' in same scope
- src/sprite.c:384: warning: `posX' might be used uninitialized in this function
- src/sprite.c:384: warning: `posY' might be used uninitialized in this function
- src/sprite.c:384: warning: `posZ' might be used uninitialized in this function
- src/sprite.c:385: warning: `rotX' might be used uninitialized in this function
- src/sprite.c:385: warning: `rotY' might be used uninitialized in this function
- src/sprite.c:385: warning: `rotZ' might be used uninitialized in this function

File: src/891b0_len_fb0.c [hunk 1]
```diff
@@ -149,13 +148,59 @@ void func_800F0864(void) {
...
+          playerStatus->flags >= 0)
+    ) {
+        D_801094AC = 4;
+        return;
+    }
 
-INCLUDE_ASM(s32, "891b0_len_fb0", func_800F0B3C);
+    if (D_801094AC++ >= 4) {
+        D_801094AC = 0;
+        sin_cos_rad((clamp_angle(-playerStatus->currentYaw) * TAU) / 360.0f, &sin, &cos);
```
Comment (Brotenko) [inline src/891b0_len_fb0.c:168]: Should use `DEG_TO_RAD` introduced in #773 for these 
```suggestion
        sin_cos_rad(clamp_angle(DEG_TO_RAD(-playerStatus->currentYaw), &sin, &cos);
```

**Review by nanaian — APPROVED** [review]

File: src/os/39cb0_len_100.c [hunk 2]
```diff
@@ -7,7 +7,7 @@
 extern u64 nuScStack[NU_SC_STACK_SIZE / sizeof(u64)];
 
 // probably should be at 0x8009A630, unless D_8009A630 (what's technically used in boot_idle) isn't nuIdleFunc
-BSS void (*nuIdleFunc)(void);
+extern void (*nuIdleFunc)(void);
```
Comment (nanaian) [inline src/os/39cb0_len_100.c:10]: This produces a warning because its alr declared

**Review by Brotenko — APPROVED** [review]
gord job!

File: src/os/39cb0_len_100.c (hunk 2 above)
Comment (ethteck) [reply src/os/39cb0_len_100.c:10]: we'll want to have this here anyway once BSS is sorted out, so I think the warning is justified

**Pull request merged**